### PR TITLE
Issue 1154 wrong nested types conversion

### PR DESCRIPTION
**Pull request by gaschd**
### Problem
https://github.com/icsharpcode/CodeConverter/issues/1154

Namespaces and types can have the same name with a different casing:
```vbnet
Namespace CaseSensitive1
    Public Class Casesensitive1
        Public Class TestDummyAttribute
            Inherits Attribute
        End Class
    End Class
End Namespace

Namespace Issue1154
    <CaseSensitive1.Casesensitive1.TestDummy>
    Public Class UpperLowerCase
    End Class
End Namespace
```
The vbnode-to-csharpnode replacement searches node names with "StringComparison.OrdinalIgnoreCase".
Due to this way of comparison, the type name will be replaced with the namespace name.

### Solution
Performing the node name search with "StringComparison.Ordinal" first. If no match is found retry with "StringComparison.OrdinalIgnoreCase". IgnoreCase is needed as 100's of tests fail when just doing Ordinal comparison.

It is not optimal that the search is done twice now for a lot of declarations, just to handle this edge case, but checking for duplicates with different casing is probably even slower, as the collections are usually very short.

### CODE REVIEW

File: CodeConverter/CSharp/CommonConversions.cs [hunk 1]
```diff
@@ -229,7 +229,9 @@ private static TypeSyntax WithDeclarationNameCasing(TypeSyntax syntax, ITypeSymb
 
         return syntax.ReplaceNodes(syntax.DescendantNodes().OfType<CSSyntax.IdentifierNameSyntax>(), (oldNode, _) =>
         {
-            var originalName = originalNames.FirstOrDefault(on => string.Equals(on, oldNode.ToString(), StringComparison.OrdinalIgnoreCase));
+            string oldNodeStr = oldNode.ToString();
```
Comment (GrahamTheCoder) [inline CodeConverter/CSharp/CommonConversions.cs:232]: Existing issue - doesn't need to be fixed here:
It's a bit odd that this uses ToString on the nodes rather than using the identifier text directly. It's _probably_ the same in the implementation, but it's weird this relies on it.

**Comment by GrahamTheCoder** [comment]
Thanks!
